bgpd/bgp_packet.c

```diff
@@ -1197,10 +1205,17 @@ static int bgp_open_receive(struct peer *peer, bgp_size_t size)
 		}
 	}
 
-	/* remote router-id check. */
+	/* rfc6286:
+	 * If the BGP Identifier field of the OPEN message
+	 * is zero, or if it is the same as the BGP Identifier
+	 * of the local BGP speaker and the message is from an
+	 * internal peer, then the Error Subcode is set to
+	 * "Bad BGP Identifier".
+	 */
 	if (remote_id.s_addr == INADDR_ANY
 	    || IPV4_CLASS_DE(ntohl(remote_id.s_addr))
-	    || ntohl(peer->local_id.s_addr) == ntohl(remote_id.s_addr)) {
+	    || (peer->sort == BGP_PEER_IBGP
+		&& ntohl(peer->local_id.s_addr) == ntohl(remote_id.s_addr))) {
 		if (bgp_debug_neighbor_events(peer))
 			zlog_debug("%s bad OPEN, wrong router identifier %s",
 				   peer->host, inet_ntoa(remote_id));
```

Comment:
In function bgp_zebra_nexthop_set we are using local_id from peer structure. 

		ret = if_get_ipv4_address(ifp, &nexthop->v4);
		if (!ret && peer->local_id.s_addr != INADDR_ANY)
			nexthop->v4 = peer->local_id;

With this feature enabled will not cause issue with nexthop being set to wrong value?

Reply:
I'm trying to generate some situations in my head with this in mind, but can't tell you quickly.

Reply:
I can't think of any situation where this would cause a problem, so I think it's okay.

Reply:
So this seems to be a dead code here?